### ISSUE TITLE
HTTPS Support?

### ISSUE THREAD

**Issue by louy2** [chriskrycho/newrustacean.com#58]
With a custom domain this is ultimately blocked on https://github.com/isaacs/github/issues/156 .

It is controversial whether to use a CDN to do partial HTTPS such as CloudFlare's Flexible SSL.

**Comment by chriskrycho** [comment]
Yeah, I've thought about alternatives for that reason for a while, for that reason. I have some hope that GitHub will push out a solution sometime this year, since they now support SSL for `<whatever>.github.io` domains.

I'm also toying a bit with the idea of using something like [PubStorm](https://www.pubstorm.com), which has free SSL hosting etc. Unfortunately, right now it's a relatively low priority (because I'm swamped with other stuff until October or so, and that would entail tweaking the publishing flow which isn't _huge_ but does take some time). If you see anything else that's a good fit, let me know. Otherwise, whenever GitHub updates to support this, I'll flip over post-haste.

Leaving this open as a standing reminder to myself. 😉

**Comment by louy2** [comment]
There are many similar to PubStorm, the earliest one I came across was [Surge](https://surge.sh/pricing), but custom domain TLS is only for paid tier. By googling [Netlify](https://www.netlify.com/pricing/) came out pretty high in the list and NewRustacean may qualify as open source thus getting CDN at Pro tier for free. They are still pretty much PubStorm.

**Comment by chriskrycho** [comment]
I just finished setting up TLS for [Mass Affection](https://www.massaffection.com) via GitLab Pages. Now that I have a handle on the process, I'm likely to migrate this whole site over there, seeing as GitHub doesn't seem interested in supporting TLS for CNAME-hosted domains. (I may leave this repository as a mirror, however, to let people contribute wherever is convenient.)

**Comment by Rukenshia** [comment]
maybe another solution would be just throwing the output on an AWS S3 Bucket and exposing it either directly or via CloudFront. AWS offers free SSL Certificates. I'm doing this for one of my websites right now and effective costs are about 60ct per month (this does not include prices when not having the 'free tier' of AWS).

**Comment by chriskrycho** [comment]
@Rukenshia not a bad suggestion. I've chosen for various reasons not to use any of Amazon's services at this point, but I may see if there are any similar options on any of their (non-megacorporation) competitors.

**Comment by moshez** [comment]
FYI -- Setting up CloudFlare you can use full HTTPS by using https://customdomain.com --> https://whatever.github.io and that falls inside the free tier. The only annoying thing is having to explicitly purge CloudFlare on publish

**Comment by chriskrycho** [comment]
Will be switching to Netlify either later this week or early next week. Super handy for this, if anyone is curious – I just have to check a couple things on the build side to make sure it'll all play nicely.

**Comment by chriskrycho** [comment]
[Done](https://newrustacean.com)… if the name servers will ever totally settle down. Seeing a lot of weirdness there. 😢
